### ISSUE TITLE
STM32CubeIDE compatibility

### ISSUE THREAD

**Issue by julian-poidevin** [de-jcup/eclipse-batch-editor#16]
The plugin does not seems to work with [STM32CubeIDE](https://www.st.com/en/development-tools/stm32cubeide.html).

![image](https://user-images.githubusercontent.com/8756421/87515992-f8c31400-c67c-11ea-88af-cdf242cc807e.png)

Which type of logs can I provide so it could help you solve the issue ?

Thanks !

**Comment by de-jcup** [comment]
Hmm.. I am not very familiar with this IDE, but being eclipsed based I try to give you some hints:

1. As looking at the icon showin up in the tab, this is NOT Batch Editor showing  up, maybe generic text editor? (Batch Editor has got its own icon - look to next picture were gradlew batch script is opened by Batch Editor)

![image](https://user-images.githubusercontent.com/11644753/88439434-833d0d80-ce0b-11ea-8afe-049c1369ef95.png)

2. Check if Batch-Editor is defined as your default editor 
    Open Window-->Preferences and search for file as shown in next picture:
![image](https://user-images.githubusercontent.com/11644753/88439797-82f14200-ce0c-11ea-9c25-2857d9353875.png)

3. If you cannot add Batch-Editor as default because its not listed,
    check if Batch-Editor is installed correctly by opening
    Help->About and press "Installation details" button + search for "batch"
![image](https://user-images.githubusercontent.com/11644753/88439926-d2d00900-ce0c-11ea-9668-6d1f5c5806ea.png)

If it is not listed here, your installation failed or startup of plugin was not working

4. Check if startup failed
Open `Window -> Show view -> Error Log` and search for entries about the plugin. If there is too much noise and you are overwhelmed by entries press "Delete Log" button (will permanently delete all logs) and do a restart of your eclipse installation. After restart is done, look into error view list, if there are entries about Batch-Editor problems etc.

5. If there are still problems, then...
- Provide more information about JRE , eclipse runtime (see Help->About Eclipse text information)
- Append logs etc. to this issue when found

**Comment by de-jcup** [comment]
I also added those information now to 
https://github.com/de-jcup/eclipse-batch-editor/wiki/FAQ#my-bat-files-are-not-opened-with-batch-editor

**Comment by julian-poidevin** [comment]
Thanks a lot for those detailed instructions, unfortunately, everything things configured the way it is on the screenshots you provided :

- [x] Batch-Editor is defined as my default editor for `*.bat` files :
![image](https://user-images.githubusercontent.com/8756421/88516437-914a8400-cfed-11ea-906e-55a7a48e9291.png)

- [x] Batch-Editor is correctly installed :
![image](https://user-images.githubusercontent.com/8756421/88516534-b50dca00-cfed-11ea-8e78-2b8649290eb5.png)

- [x] Startup didn't fail :
![image](https://user-images.githubusercontent.com/8756421/88516632-e2f30e80-cfed-11ea-80e7-26b33d35abfb.png)

Here's some more information about the Eclipse configuration :

| Name | Screenshot  |
|---|---|
| Eclipse Build  | ![image](https://user-images.githubusercontent.com/8756421/88516764-10d85300-cfee-11ea-9b34-9beb904d3cb7.png)  |
| CubeIDE Version and Build  | ![image](https://user-images.githubusercontent.com/8756421/88517021-7c222500-cfee-11ea-9b15-c6be60fecac0.png)  |
| Installation details (part 1)  | ![image](https://user-images.githubusercontent.com/8756421/88517244-da4f0800-cfee-11ea-84e3-ea0272906e55.png)  |
| Installation details (part 2)  | ![image](https://user-images.githubusercontent.com/8756421/88517312-f0f55f00-cfee-11ea-830f-facdde4d1f95.png) |
|Installation details (part 3) | ![image](https://user-images.githubusercontent.com/8756421/88517411-15513b80-cfef-11ea-8308-fc7a5f14f7d6.png) |
|Eclipse Startup log | [startup.log](https://github.com/de-jcup/eclipse-batch-editor/files/4980680/startup.log) |


Tell me if I can provide more information, thanks a lot for your support, really appreciated

**Comment by julian-poidevin** [comment]
Hello @de-jcup, does the information provided could help you ?
Thanks,

**Comment by de-jcup** [comment]
Hello @julian-poidevin , Looking at your pictures, installation seems to be okay, but the icon shown for opened editor is not the icon for "Batch Editor", so I am sure the wrong editor appears - the file associations in your picture showing your preferences does show up correct editor (and icon).

Have you tried to select your file in project explorer/navigator, then open context menu  and execute 
`Open with ...` 
and select explicit "Batch Editor" ? This does explicit set batch editor as the editor to edit THIS file.

**Comment by julian-poidevin** [comment]
Works great thanks. I never tried to open the file with the `Open with ...` option, always double clicked on the file.

Thanks a lot !

**Comment by de-jcup** [comment]
Nice to here that it works now.

**tl;dr: After having done this one time, a doubleclick on this file should work now as well!**

_Explanation what happend (IMHO):_
You must have opened the file formerly with another editor (using accidently `Open with ...`) - this information is stored inside eclipse workspace and does override `default` file association. 

So at next time you do a doubleclick on your mentioned file,  eclipse knows which editor was last explicit set (you have done this now to `Batch Editor`...) and will do this automatically for you - means there should be no longer a need for `Open with...` at this file)
